test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -803,6 +804,32 @@ UCS_TEST_P(test_ucp_worker_address_query, query)
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_worker_address_query)
 
+class test_ucp_worker_address_version : public ucp_test {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_AM, 0, "");
+    }
+};
+
+UCS_TEST_P(test_ucp_worker_address_version, pack_unpack)
+{
+    uint8_t v17_packed     = ucp_address_packed_release_version(17u);
+    uint8_t v18_packed     = ucp_address_packed_release_version(18u);
+    uint8_t current_packed = ucp_address_pack_release_version();
+
+    unsigned v17_unpacked     = ucp_address_unpack_release_version(v17_packed);
+    unsigned v18_unpacked     = ucp_address_unpack_release_version(v18_packed);
+    unsigned current_unpacked = ucp_address_unpack_release_version(current_packed);
+
+    ASSERT_EQ(17u, v17_unpacked);
+    ASSERT_EQ(18u, v18_unpacked);
+    ASSERT_EQ(UCP_API_MINOR, current_unpacked);
+    EXPECT_LT(v18_unpacked, current_unpacked);
```

Comment:
this  will fail once the version is bumped

Reply:
btw, should not it fail already, because the current minor is 19?
why would we need it to fail for every new minor release version?

Reply:
> btw, should not it fail already, because the current minor is 19?

no, 18 < 19
> why would we need it to fail for every new minor release version?

it wont fail every minor release version, only 17 + 4, then we can bump legacy version in the test as needed.

Reply:
right, it is `LT`, not `EQ`. So it wil lfail in 8 releases